packages/bridge/contracts/l2/L2DaiGateway.sol

```diff
@@ -78,49 +96,94 @@ contract L2DaiGateway is L2ArbitrumGateway {
     emit Closed();
   }
 
-  function handleNoContract(
-    address l1ERC20,
-    address expectedL2Address,
-    address _from,
-    address _to,
-    uint256 _amount,
-    bytes memory gatewayData
-  ) internal virtual override returns (bool shouldHalt) {
-    // it is assumed that the custom token is deployed in the L2 before deposits are made
-    // trigger withdrawal
-    createOutboundTx(_from, _amount, gatewayData);
-    return true;
+  function outboundTransfer(
+    address l1Token,
+    address to,
+    uint256 amount,
+    bytes calldata data
+  ) public virtual returns (bytes memory) {
+    return outboundTransfer(l1Token, to, amount, 0, 0, data);
   }
 
-  function calculateL2TokenAddress(address l1ERC20) public view virtual override returns (address) {
```

Comment:
This function is part of the token gateway interface https://github.com/OffchainLabs/arbitrum/blob/master/packages/arb-bridge-peripherals/contracts/tokenbridge/libraries/gateway/ITokenGateway.sol#L66

It is used to validate if a gateway can properly handle a given token.
Should be as simple as returning l2Dai if the input is l1Dai, or address(0) if l1ERC20 is not l1Dai.